app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/MainActivity.kt

```diff
@@ -0,0 +1,493 @@
+                        BUNDLE_KEY_USE_DEVICE_TIME_ZONE_UPDATED, false)
+
+                    if (isAlternativeHighlightingUpdated || isUseDeviceTimeZoneUpdated) {
+                        if (findViewById<View>(R.id.schedule) != null && findFragment(FahrplanFragment.FRAGMENT_TAG) == null) {
+                            replaceFragment(R.id.schedule, FahrplanFragment(), FahrplanFragment.FRAGMENT_TAG)
+                        }
+                    }
+                    var shouldFetchFahrplan = false
+                    val isScheduleUrlUpdated = resources.getBoolean(
+                        R.bool.bundle_key_schedule_url_updated_default_value)
+                    if (intent.getBooleanExtra(BUNDLE_KEY_SCHEDULE_URL_UPDATED, isScheduleUrlUpdated)) {
+                        shouldFetchFahrplan = true
+                    }
+                    val isEngelsystemShiftsUrlUpdated = resources.getBoolean(
+                        R.bool.bundle_key_engelsystem_shifts_url_updated_default_value)
+                    if (intent.getBooleanExtra(BUNDLE_KEY_ENGELSYSTEM_SHIFTS_URL_UPDATED, isEngelsystemShiftsUrlUpdated)) {
+                        shouldFetchFahrplan = true
+                    }
+                    if (shouldFetchFahrplan) {
+                        fetchFahrplan()
+                    }
+                }
+        }
+    }
+
+    override fun onSessionListClick(sessionId: String) {
+        openSessionDetails(sessionId)
+    }
+
+    override fun onSessionClick(sessionId: String) {
+        openSessionDetails(sessionId)
+    }
+
+    override fun onBackStackChanged() {
+        toggleSidePaneViewVisibility(supportFragmentManager, R.id.detail)
+        invalidateOptionsMenu()
+    }
+
+    private fun showProgressDialog(@StringRes message: Int) {
+        progressDialog = ProgressDialog.show(this, "", resources.getString(message), true)
+    }
+
+    private fun hideProgressDialog() {
+        progressDialog?.let {
+            it.dismiss()
+            resetProgressDialog()
+        }
+    }
+
+    private fun resetProgressDialog() {
+        progressDialog = null
+    }
+
+    private fun toggleSidePaneViewVisibility(fragmentManager: FragmentManager, @IdRes detailView: Int) {
+        val fragment = fragmentManager.findFragmentById(detailView)
+        val found = fragment != null
+        findViewById<View>(detailView)?.let { view ->
+            isFavoritesInSidePane = found && fragment is StarredListFragment
+            view.isVisible = !(isFavoritesInSidePane && isScreenLocked || !found)
+        }
+    }
+
+    private fun openFavorites() {
+        val sidePaneView = findViewById<FragmentContainerView>(R.id.detail)
+        if (sidePaneView == null) {
+            StarredListActivity.start(this)
+        } else if (!isScreenLocked) {
+            sidePaneView.isVisible = true
+            isFavoritesInSidePane = true
+            StarredListFragment.replace(supportFragmentManager, R.id.detail, true)
+        }
```

Comment:
Not sure if `when` is more programmatic:
```
when {
    sidePaneView !is View -> StarredListActivity.start(this)
    !isScreenLocked -> {
        sidePaneView.isVisible = true
        isFavoritesInSidePane = true
        StarredListFragment.replace(supportFragmentManager, R.id.detail, true)
    }
}
```

Reply:
:x: Not much of a difference I'd say.